src/api/GameContext.tsx

```diff
@@ -26,24 +39,68 @@ type DeleteGameAction = {
   type: 'DELETE_GAME'
 }
 
-type Action = SetCardsAction | DeleteGameAction
+type FinishGameAction = {
+  type: 'FINISH_GAME'
+}
+
+type UpdateButtonAction = {
+  type: 'UPDATE_BUTTON'
+}
+
+type Action =
+  | StartGameAction
+  | SetCardIndexAction
+  | SetCardsAction
+  | DeleteGameAction
+  | FinishGameAction
+  | UpdateButtonAction
 
 type Props = {
   state: State
   dispatch: React.Dispatch<Action>
+  handleButtonClick: () => void
 }
 
 const initialState: State = {
   cards: [],
-  isGameOngoing: false,
+  gameState: GameState.NOT_STARTED,
+  buttonLabel: 'New Game',
+  currentCardIndex: 0,
 }
 
 const gameReducer = (state: State, action: Action): State => {
   switch (action.type) {
+    case 'START_GAME':
+      return { ...state, gameState: GameState.START, currentCardIndex: 0 }
```

Comment:
Macht für mich nur bedingt Sinn, weil wenn wir ein Game starten ist es eigentlich automatisch ongoing. Vielleicht etwas overkill in der Logik?

Reply:
done in #41